### PR TITLE
Feature/userdefaults helper and constants

### PR DESCRIPTION
**Pull request by onurfeyyaz**
Added UserDefaults Helper class.
Created Constants file for managing constants from one place.

### CODE REVIEW

File: Things To Do/Scenes/Home/HomeViewModel.swift [hunk 1]
```diff
@@ -7,35 +7,38 @@
...
         }
     }
-    // didSet?
     
-    private let userDefaultsKey = "toDoItemsKey"
+    private let userDefaultsHelper: UserDefaultsHelper<ToDoItem>
     
-    init() {
-        loadFromUserDefaults()
+    init(userDefaultsHelper: UserDefaultsHelper<ToDoItem>) {
```
Comment (beyzaince) [inline Things To Do/Scenes/Home/HomeViewModel.swift]: init(userDefaultsHelper: UserDefaultsHelper<ToDoItem> = UserDefaultsHelper<ToDoItem>(userDefaultsKey: Constants.toDoItemsKey))

File: Things To Do/Utilities/UserDefaultsHelper.swift [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        } catch {
+            #if DEBUG
+                print("UserDefaults encoding error: \(error)")
+            #else
+                //TODO: show generic alert view
+            #endif
+        }
+    }
+    
+    func loadToDoItems() -> [T] {
```
Comment (beyzaince) [inline Things To Do/Utilities/UserDefaultsHelper.swift]: more generic class definiton

File: Things To Do/Scenes/Home/HomeViewModel.swift (hunk 1 above)
Comment (onurfeyyaz) [reply Things To Do/Scenes/Home/HomeViewModel.swift]: thanks, did it

File: Things To Do/Utilities/UserDefaultsHelper.swift (hunk 2 above)
Comment (onurfeyyaz) [reply Things To Do/Utilities/UserDefaultsHelper.swift]: thanks, i changed save and load functions names to make more generic.

File: Things To Do/Scenes/Home/HomeViewModel.swift [hunk 3]
```diff
@@ -7,35 +7,35 @@
...
     
-    func addToDoItem(title: String) {
-        let newItem = ToDoItem(title: title, isCompleted: false)
-        toDoItems.append(newItem)
+    init(userDefaultsHelper: UserDefaultsHelper<ToDoItem> = UserDefaultsHelper<ToDoItem>(userDefaultsKey: Constants.toDoItemsKey)) {
+        self.userDefaultsHelper = userDefaultsHelper
+        self.toDoItems = loadFromUserDefaults()
     }
     
     private func saveToUserDefaults() {
```
Comment (beyzaince) [inline Things To Do/Scenes/Home/HomeViewModel.swift]: rm

File: Things To Do/Utilities/UserDefaultsHelper.swift [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+//  Things To Do
+//
+//  Created by Feyyaz ONUR on 2.07.2024.
+//
+
+import Foundation
+
+protocol UserDefaultsHelperProtocol {
+    associatedtype T: Codable
+    func saveData(_ items: [T])
```
Comment (beyzaince) [inline Things To Do/Utilities/UserDefaultsHelper.swift:12]: .

File: Things To Do/Scenes/Home/HomeViewModel.swift (hunk 3 above)
Comment (onurfeyyaz) [reply Things To Do/Scenes/Home/HomeViewModel.swift]: done!

File: Things To Do/Utilities/UserDefaultsHelper.swift (hunk 4 above)
Comment (onurfeyyaz) [reply Things To Do/Utilities/UserDefaultsHelper.swift:12]: when i change [T] to T, it gives me error at ViewModel like "Cannot convert return expression of type 'ToDoItem' to return type '[ToDoItem]'". so i guess T does not include [T]. and i found a usage of [T] in swift docs: https://docs.swift.org/swift-book/documentation/the-swift-programming-language/generics/

**Pull request merged**